src/java.base/share/classes/java/lang/ref/NativeReferenceQueue.java

```diff
@@ -66,7 +66,7 @@ public Reference<? extends T> poll() {
             return null;
 
         synchronized(lock) {
-            return poll0();
+            return super.poll();
```

Comment:
headIsNull() is not implemented either.

As it stands there are two sets of locks, the one implemented here and the one used in OpenJ9 ReferenceQueue. Given the description of NativeReferenceQueue, I think it's better to delete the method implementations and inherit them, rather than re-implementing them with an extra lock we don't need.